AndrewTehan/hometask-3/rainfall_statistics.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# this class is used for calculate mean and variance values for town rainfall
+class RainfallStatistics
+  attr_reader :town, :data
+
+  def initialize(town, data)
+    @town = town
+    @data = data
+  end
+
+  def mean
+    return -1 if missing_data?
+
+    average(extract_statistic_values)
+  end
+
+  def variance
+    return -1 if missing_data?
+
+    calculate_variance(calculate_mean_differences)
+  end
+
+  private
+
+  def extract_statistic_values
+    @extract_statistic_values ||= find_town_statistics.scan(/[0-9]*\.[0-9]*/).map(&:to_f)
+  end
+
+  def missing_data?
+    !find_town_statistics
+  end
+
+  def find_town_statistics
+    @find_town_statistics ||= data.each_line.find do |line|
+      line.start_with?(town)
+    end
+  end
+
+  def calculate_mean_differences
+    extract_statistic_values.map { |town_monthly_rainfall| town_monthly_rainfall - mean }
+  end
+
+  def average(values)
+    values.sum(&:to_f) / values.size
+  end
+
+  def calculate_variance(values)
+    values.sum { |value| value**2 } / values.size
+  end
```

Comment:
I'm wondering why methods are called according to different concepts. They both do some calculations, but one of them is called `calculate_*` and another one is just `average`

So, it's not so critical there, but it's better to start to control such stuff even in small programs.